### ISSUE TITLE
Weird bus error

### ISSUE THREAD

**Issue by rcqls** [vlang/v#14166]
V 0.2.4 501b293
OS: macos, macOS, 11.6, 20G165

**What did you do?**

```v
[heap]
struct Menu {
	items []MenuItem
}

struct MenuItem {
	text    string
	submenu &Menu = 0
}

fn menuitem(mi MenuItem) MenuItem {
	return mi
}

fn menu(m &Menu) &Menu {
	return m
}

fn main() {
	items := [
		menuitem(
			text: 'Delete'
			submenu: menu(
				items: [
					menuitem(
						text: 'users'
						submenu: menu(
							items: [
								menuitem(
									text: 'devel'
									submenu: menu(
										items: [
											menuitem(
												text: 'devel'
												// COMMENTING LINE BELOW WORKS
												submenu: menu(items: []MenuItem{})
											),
										]
									)
								),
							]
						)
					),
				]
			)
		),
	]
}
```

**What did you expect to see?**
No error like when I comment the deepest instruction `submenu(...)` in the previous code
 
**What did you see instead?**

Bus error

**Comment by rcqls** [comment]
```v
fn f(x int) int {
        return x
}

fn main() {
        f(f(f(f(f(f(f(f(f(f(f(f(f(f(f(f(2))))))))))))))))
}
```
is a shorter example.

**Comment by yuyi98** [comment]
It's ok on win11.
```v
[heap]
struct Menu {
	items []MenuItem
}

struct MenuItem {
	text    string
	submenu &Menu = 0
}

fn menuitem(mi MenuItem) MenuItem {
	return mi
}

fn menu(m &Menu) &Menu {
	return m
}

fn main() {
	items := [
		menuitem(
			text: 'Delete'
			submenu: menu(
				items: [
					menuitem(
						text: 'users'
						submenu: menu(
							items: [
								menuitem(
									text: 'devel'
									submenu: menu(
										items: [
											menuitem(
												text: 'devel'
												// COMMENTING LINE BELOW WORKS
												submenu: menu(items: []MenuItem{})
											),
										]
									)
								),
							]
						)
					),
				]
			)
		),
	]
	println(items)
}

PS D:\Test\v\tt1> v run .
[MenuItem{
    text: 'Delete'
    submenu: &Menu{
        items: [MenuItem{
            text: 'users'
            submenu: &Menu{
                items: [MenuItem{
                    text: 'devel'
                    submenu: &Menu{
                        items: [MenuItem{
                            text: 'devel'
                            submenu: &Menu{
                                items: []
                            }
                        }]
                    }
                }]
            }
        }]
    }
}]
```
```v
fn f(x int) int {
	return x
}

fn main() {
	ret := f(f(f(f(f(f(f(f(f(f(f(f(f(f(f(f(2))))))))))))))))
	println(ret)
}

PS D:\Test\v\tt1> v run .
2
```

**Comment by Hunam6** [comment]
I've just tested on MacOS, Windows and Linux: it's a MacOS specific issue

**Comment by rcqls** [comment]
> I've just tested on MacOS, Windows and Linux: it's a MacOS specific issue

And @spytheman found an explanation.

**Comment by dumblob** [comment]
Any recommended workarounds?

**Comment by Hunam6** [comment]
> Any recommended workarounds?

@spytheman suggested to compile with `-no-parallel` (https://canary.discord.com/channels/592103645835821068/592114487759470596/968194895586132058)

**Comment by rcqls** [comment]
I think @spytheman already proposed a PR to fix that (by increasing the mempry for pthreads). I have no more this error.

**Comment by Hunam6** [comment]
That what I thought too but he didn't linked this issue, @spytheman?

**Comment by spytheman** [comment]
Sorry, I should have closed it. It was fixed in be04ec0 . The default thread stack size is now 8MB everywhere, it can be changed with `-thread-stack-size 4194304` for example.
